src/main/java/com/internet/shop/dao/ShoppingCartDaoImpl.java

```diff
@@ -33,6 +41,11 @@ public ShoppingCart update(ShoppingCart shoppingCart) {
         return shoppingCart;
     }
 
+    @Override
+    public boolean deleteById(Long itemId) {
+        return Storage.shoppingCarts.removeIf(s -> s.getId().equals(itemId));
+    }
+
     @Override
     public boolean delete(ShoppingCart shoppingCart) {
```

Comment:
i believe deleteById() would be enough, since in this method you just repeat same logic

Reply:
done